sync/sync_head.go

```diff
@@ -186,7 +193,68 @@ func (s *Syncer[H]) verify(ctx context.Context, newHead H) (bool, error) {
 			"reason", verErr.Reason)
 	}
 
-	return verErr.SoftFailure, err
+	return false, err
+}
+
+// verifySkipping will try to find such headers in range (subjHead, networkHeader)
+// that can be verified by subjHead, literally:
+//
+//	header.Verify(subjHead, candidate)
+//
+// and also such headers can verify `networkHeader`, literally
+//
+//	header.Verify(candidate, networkHeader)
+//
+// When such candidates cannot be found [NewValidatorSetCantBeTrustedError] will be returned.
+func (s *Syncer[H]) verifySkipping(ctx context.Context, subjHead, networkHeader H) error {
+	subjHeight := subjHead.Height()
+
+	diff := networkHeader.Height() - subjHeight
+	if diff <= 0 {
+		panic(fmt.Sprintf("implementation bug: diff is %d", diff))
```

Comment:
let's print the network head height and the subjective head height here too as well as their hashes

Reply:
Done